### PR TITLE
[CI][Misc] Upgrade vllm to v0.28.0

### PR DESCRIPTION
**Pull request by LQDLove**
### What this PR does / why we need it?

main2main vllm upgrade to v0.28.0

---

Update CI tracking tag: `.github/vllm-release-tag.commit` from `v0.27.1` to `v0.28.0`, and adapt the codebase so vllm-ascend works on both vLLM v0.28.0 and vLLM main.

#### 1. Version / CI / Docker / Documentation

| File | Change Reason |
|------|---------------|
| `.github/vllm-release-tag.commit` | Update CI tracking tag: `v0.27.1` → `v0.28.0` |
| `Dockerfile`, `Dockerfile.310p`, `Dockerfile.310p.openEuler`, `Dockerfile.a3`, `Dockerfile.a3.openEuler`, `Dockerfile.a5`, `Dockerfile.a5.openEuler`, `Dockerfile.openEuler` | `ARG VLLM_TAG=v0.27.1` → `v0.28.0` |
| `README.md`, `README.zh.md` | Update main branch description: CI commitment for vLLM `v0.27.1` → `v0.28.0` tag |

#### 2. Drop v0.27.1 version gates (v0.28.0 upstream matches main)

All `vllm_version_is("0.27.1")` branches whose behavior v0.28.0 no longer needs were removed, keeping only the current (main) code paths:

| File | Change Reason |
|------|---------------|
| `vllm_ascend/worker/v2/model_runner.py` | Unconditional `BatchReqState` import; `execute_model` always passes `context_len`; single `prepare_inputs` with the main signature |
| `vllm_ascend/worker/v2/input_batch.py` | Single dataclass layout with the `max_query_len` default field |
| `vllm_ascend/worker/v2/aclgraph_utils.py` | Keep the main `__init__` signature only |
| `vllm_ascend/worker/model_runner_v1.py` | Drop routed-experts `clear_buffer()` and `calculate_kv_scales` gates (removed on main only, vllm #50721/#49389) |
| `vllm_ascend/_310p/fused_moe/grouped_topk_router.py` | Always handle fp32-recomputed router_logits (FRACTAL_NZ on 310P) |
| `vllm_ascend/models/deepseek_mtp.py` | Always load with `AutoWeightsLoader(skip_prefixes=["rot."])` |
| `vllm_ascend/models/qwen3_dspark.py` | Drop the local `DSparkConfidenceHead`; upstream provides it (vllm #47808) |
| `vllm_ascend/ops/fused_moe/fused_moe.py` | Keep only the main `weight_fp32` gate branch (vllm #51838) |
| `vllm_ascend/ops/triton/mamba/postprocess.py` | Keep only the main `postprocess_mamba_fused_kernel` |
| `vllm_ascend/patch/platform/patch_kv_cache_coordinator.py` | Keep only the main `HybridKVCacheCoordinator` signature |
| `vllm_ascend/patch/worker/patch_qwen3_5.py` | Keep only the main qwen3_next `_all_gather_hidden_and_residual` usage |
| `vllm_ascend/compilation/acl_graph.py` | Keep only the main `get_current_vllm_config()` handling |
| `tests/ut/patch/platform/test_patch_structured_output.py` | Normalize to v0.28.0: `compile_grammar` takes `stop_token_ids`, `_create_grammar` reads `all_stop_token_ids`, validation raises `VLLMValidationError` |
| `tests/ut/worker/test_model_runner_v2.py` | Drop the `is_vllm_0_27_1` parametrization; `context_len` is always passed |
| `tests/ut/patch/worker/test_patch_mamba_utils_source.py` | Expect the single (main) postprocess kernel |
| `tests/ut/kv_offload/test_native_cpu_offload.py`, `tests/ut/ops/test_routed_experts.py`, `tests/ut/patch/platform/test_deepseek_v4_thinking.py`, `tests/ut/spec_decode/test_extract_hidden_states_proposer.py`, `tests/e2e/pull_request/two_card/rlhf/consistency/test_moe_routing_replay.py` | Drop 0.27.1 branches, keep main behavior |

#### 3. Bump version gates to v0.28.0 (v0.28.0 still needs the gated behavior)

| File | Change Reason |
|------|---------------|
| `vllm_ascend/attention/attention_v1.py` | `_gather_prefill_cache_inputs` lives in `vllm.model_executor.layers.attention.pcp` at v0.28.0, moved to `vllm.v1.attention.ops.pcp` on main |
| `vllm_ascend/worker/v2/spec_decode/dflash/speculator.py` | Gate the dflash kernel variant with temperature/seeds parameters (vllm #50000) |
| `vllm_ascend/spec_decode/llm_base_proposer.py` | v0.28.0 MTP architecture set (`DeepSeekMTPModel`/`KimiK3MTPModel`) |
| `tests/ut/core/test_recompute_scheduler.py`, `tests/ut/core/test_profiling_chunk.py`, `tests/e2e/conftest.py` | Update version gates from 0.27.1 to 0.28.0 |

#### 4. Remove patch_vision FusedInputNorm patch

| File | Change Reason |
|------|---------------|
| `vllm_ascend/patch/platform/patch_vision.py` (deleted), `vllm_ascend/patch/platform/__init__.py`, `vllm_ascend/patch/__init__.py` | Upstream vllm #51734 (included in v0.28.0) rewrote `FusedInputNorm.forward` to a broadcast multiply-add, removing `F.batch_norm`; the eps=0.0 → eps=1e-5 workaround is no longer needed |

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

CI verification only

---

- vLLM main: https://github.com/vllm-project/vllm/commit/b2f685834a6456197e7033966fdef52a23f1abcd

### CODE REVIEW

**Comment by LQDLove** [comment]
/cancel
[Bot]: successfully force-cancelled the following runs:


https://github.com/vllm-project/vllm-ascend/actions/runs/32842456624

**Comment by LQDLove** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by LQDLove** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

File: vllm_ascend/ops/fused_moe/fused_moe.py [hunk 1]
```diff
@@ -204,106 +199,59 @@ def set_lora_context(self, lora_context):
...
+                # linear is unquantized so that we the weight is pre-casted in
+                # process_weights_after_loading of AscendUnquantizedLinearMethod.
+                hidden_states_fp32 = hidden_states.float()
+                before_routed_experts = torch.npu.current_stream().record_event()
+                # main (cdc4824a21): is_internal_router only checks self.gate,
+                # weight_fp32 may be absent, fall back to gate.weight.
+                router_logits = F.linear(
+                    hidden_states_fp32,
+                    gate.weight_fp32 if hasattr(gate, "weight_fp32") else gate.weight.to(torch.float32),
                 )
```
[the remark this answers is not in the document]
Comment (LQDLove) [reply vllm_ascend/ops/fused_moe/fused_moe.py:255]: ok

File: vllm_ascend/ops/fused_moe/fused_moe.py [hunk 2]
```diff
@@ -204,106 +199,59 @@ def set_lora_context(self, lora_context):
...
-                    # linear is unquantized so that we the weight is pre-casted in
-                    # process_weights_after_loading of AscendUnquantizedLinearMethod.
                     hidden_states_fp32 = hidden_states.float()
-                    before_routed_experts = torch.npu.current_stream().record_event()
-                    # main (cdc4824a21): is_internal_router only checks self.gate,
-                    # weight_fp32 may be absent, fall back to gate.weight.
                     router_logits = F.linear(
                         hidden_states_fp32,
                         gate.weight_fp32 if hasattr(gate, "weight_fp32") else gate.weight.to(torch.float32),
                     )
```
[the remark this answers is not in the document]
Comment (LQDLove) [reply vllm_ascend/ops/fused_moe/fused_moe.py:230]: ok

**Comment by LQDLove** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by LQDLove** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by LQDLove** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by LQDLove** [comment]
/e2e
[Bot]: e2e command triggered. See [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/34089403669) for details.

**Comment by LQDLove** [comment]
/cancel
[Bot]: successfully force-cancelled the following runs:


https://github.com/vllm-project/vllm-ascend/actions/runs/34090882727

**Comment by czydyy** [comment]
/nightly Qwen3.5-397B-A17B-w8a8-mtp  kimi-k2.5  deepseek-v3-2-w8a8 Qwen3-32B-QuaRot
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/34179191575)

**Review by Tflowers-0129 — APPROVED** [review]
